in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,270 @@
+\end{aligned}
+```
+
+This final value is the `base_fee_asset_per_mana` field in the L2 block header.
+
+## The cost of a transaction in the fee asset
+
+The amount of mana a transaction consumes is:
+
+```math
+\text{mana}_{tx} = \text{OVERHEAD MANA PER TX} + \sum_{op \in tx} \text{mana per operation}(op)
+```
+
+Therefore, the cost of a transaction's proposal, data publication, and verification at time $i$ is, denominated in the fee asset:
+
+```math
+\text{fee asset}_{tx} = \text{mana}_{tx} * \text{base fee asset per mana}
+```
+
+## Distribution of the fees
+
+When a transaction is included in a block in the pending chain, the L2 balance of the fee asset for the `fee_payer` of the transaction is reduced by the transactions's fee asset cost.
+
+When an epoch is proven, the fees for the epoch are paid out to each proposer for the block(s) they proposed.
+
+A percentage of each block's fee asset fees are paid to the prover of the epoch. The percentage is determined based on the quote submitted by the prover for the epoch.
+
+This quote is submitted by the prover in epoch `i+1` to prove epoch `i`, and is thus claimed/submitted to L1 by a proposer in epoch `i+1`.
+
+In order to prevent a prover bribing a proposer into accepting a quote with an extremely high fee, the maximum fee a prover can submit for an epoch proof quote is `MAXIMUM_EPOCH_PROOF_QUOTE_FEE`.
+
+## Clarifications and considerations
+
+### Why is `blobs_per_block` a constant?
+
+It makes computation easier. Otherwise block builder would need to know the number of blobs consumed by the transaction while it is building.
+
+### L1 Congestion
+
+If L1 gas spikes during an epoch, this will be reflected in the `wei_per_l1_gas` and `wei_per_l1_blob_gas` oracles and make their way into the cost.
+
+However, these oracles lag the current L1 base fees. Furthermore, the base fee computed at the time of a block being proposed may differ from the base fee at the time of the epoch being verified.
+
+Thus, it is possible for L1 gas prices to spike to a point where it is not profitable for a prover to submit their proof to L1. There are two retorts:
+1. The prover should bake in a risk premium to compensate them for this risk; it is similar to the risk that the price of fee asset relative to any other asset shifts against them.
+2. The prover can wait until the end of the epoch waiting for a less congested window to submit their proof to L1.
+
+### Not charging for DA separately
+
+By effectively charging each proposed block as though it uses BLOBS_PER_BLOCK blobs, we can simplify by not charging separately for DA, and instead add the use of the blobs directly to the block cost.
+
+Regardless, the AVM supports DA gas metering per opcode, so it would not be difficult to add in the future if we change our mind.
+
+Further, it is assumed that the proving cost will dominate the cost of the transaction, so this simplification makes implementation easier, and UX better.
```

Comment:
I might argue that in 5-10 years, the DA costs will dominate. There are provers that are already 10x faster than our public provers. So it'll be more like 100x+ faster in 5-10 years. Then DA will dominate.

Reply:
Even in the shorter-term, with the number of rollups coming online, blob costs will go up.